### ISSUE TITLE
Orgmode (.org) to Markdown (.md) not handling in-page link correctly

### ISSUE THREAD

**Issue by TRSx80** [jgm/pandoc#6917]
This is same issue we were discussing on [mailing list](https://groups.google.com/g/pandoc-discuss/c/D-8J4RGiYsk/m/g45AutiNAAAJ), I am filing issue here as requested.

Picking up conversation from that thread,

> What's more is that the conversion from Link to Emph is based on the behavior of Emacs Org exporters: invalid and unknown links are converted to emphasized text. So we can't even use a pandoc filter to fix it. We may want to change that as well.

Forgive me if I misunderstood you, but the example I provided should be a valid link in Org format (pointing to heading on same page), no?  I reproduce here:
```
[[*Setup][Setup]]
```
Or maybe you were saying because it's not implemented, is why it is considered an "invalid" link.  In other words, what you were saying here:

> Indeed, it's the Org reader that is to blame here. This case is not handled at all.

I was digging through [code in Org folder](https://github.com/jgm/pandoc/tree/master/src/Text/Pandoc/Readers/Org) yesterday with a mind toward contributing a PR, and that is sort of the way it looked to me, too.  However I cannot Haskell my way out of a wet paper bag (yet) and so could have been completely wrong.  :smile:  And neither can I contribute a PR.  :frowning_face: 

Emacs Lisp, Bash, for sure; Python usually, or even a little bit of C type languages maybe, but not Haskell.  Not yet, anyway.

So I must leave it in the hands of you wizards, for now.

Thanks for taking a look, and again for sharing pandoc.

Cheers!  :beers: 

EDIT: I should add that I am willing to financially sponsor the development of this feature, as I am much more concerned with free as in freedom rather than free as in cost and I don't expect anyone to do anything "for free."  I was looking at the GitHub sponsors and maybe that is the easiest / best way (especially if they are still matching?) however otherwise I would probably prefer a one off financial contribution.  Maybe I should email you privately?  I hope you do not consider this untoward in any way.  I try to support Free Software as much as I can, financially and otherwise.

**Comment by jgm** [comment]
This seems extremely close to issue #6916. Instead of having two issues, should we close this and add your similar case to the comments of #6916?

**Comment by TRSx80** [comment]
Wow, we must have submitted our issues at damn near the same time!  lol

I will be happy to add my case to the other issue.  If I am understanding what was discussed on the mailing list correctly, this is indeed an issue with the Org reader, as my case is not handled at all (and his seems to be very similar).

Closing this now and I will summarize briefly in the other issue.

Thanks!

**Comment by jgm** [comment]
Btw, if you do want to support financially, GitHub sponsors is the easiest way.
Since @tarleb is the one who supports the org reader, you could sponsor him.

**Comment by TRSx80** [comment]
> GitHub sponsors is the easiest way.

You know, after I wrote that I realized something.  I have taken great pains to compartmentalize my real identity away from this online persona, but most especially I don't trust Microsoft at all and giving them my credit card information (or whatever) would certainly ruin all of that.  Even as much as I would like to take advantage of their matching offer in the short term.  Which I am not even sure they are still doing?

So perhaps there is some other way.

Also, the real root cause of my problem -- converting my .org README to .md for purpose of nice automated HTML rendering on my sr.ht projects -- seems to suddenly become more solvable by another method.[0]  But now that I open my big mouth, I might need to make some contribution anyway so as not to be viewed like Lucy pulling the football away from poor Charlie Brown.  :smile: 

[0] In the meantime [Drew suddenly became amenable to a patch](https://lists.sr.ht/~sircmpwn/sr.ht-discuss/%3Cfe7aa296-9c90-463d-b4e6-50eeb7e57428%40localhost%3E#%3CC7JA2CS6U3L5.3ED1S5BHP97WE@taiga%3E) which (due to previous dealings) I really thought had a low probability of success.
